### PR TITLE
[Mono] Add more binary op intrinsics for ARM64 mini JIT

### PR DESCRIPTION
**Pull request by fanyang-mono**
This PR enabled emitting hardware intrinsics for the following API of Vector/Vector128 for ARM64 mini JIT:
- Subtract
- BitwiseAnd
- BitwiseOr
- Xor
- Operator `+` (one operand)
- Operator `-` (one operand)
- Operator `~`
- Operator `+`
- Operator `-`
- Operator `&`
- Operator `|`
- Operator `^`

Contribute to #80566

### CODE REVIEW

**Comment by fanyang-mono** [comment]
Test failure on `runtime (Build Libraries Test Run release mono linux arm64 Debug)` also happened on main.

**Review by jandupej — APPROVED** [review]
LGTM

File: src/mono/mono/mini/mini-ops.h [hunk 1]
```diff
@@ -493,6 +493,11 @@ MINI_OP(OP_RBGT_UN, "r4_bgt_un", NONE, NONE, NONE)
 MINI_OP(OP_RBLE_UN, "r4_ble_un", NONE, NONE, NONE)
 MINI_OP(OP_RBLT_UN, "r4_blt_un", NONE, NONE, NONE)
 
+MINI_OP(XBINOP_FORCEINT_AND,    "bitwise_and", IREG, IREG, IREG)
+MINI_OP(XBINOP_FORCEINT_OR,     "bitwise_or", IREG, IREG, IREG)
```
Comment (vargaz) [inline src/mono/mono/mini/mini-ops.h:497]: These are not actually opcodes, so they shouldn't be in this file.

Comment (fanyang-mono) [reply src/mono/mono/mini/mini-ops.h]: Created enum in mini.h instead.

**Comment by fanyang-mono** [comment]
CI failures on wasm are not related to this PR.

**Pull request merged**